From a430f0774dcc075110f3f277c9e7e45fd7f6f883 Mon Sep 17 00:00:00 2001
From: Ankit Jain <radical@gmail.com>
Date: Tue, 19 Feb 2019 23:31:12 -0500
Subject: [PATCH] Fix build failure with mcs caused by C#8 changes in mcs

Recent addition of C#8 features to mcs possibly broke this:

```
mcs -out:WidgetViewer.exe -r:../../glib/glib-sharp.dll -r:../../pango/pango-sharp.dll -r:../../atk/atk-sharp.dll -r:../../gdk/gdk-sharp.dll -r:../../gtk/gtk-sharp.dll ./TestCheckButton.cs ./TestColorSelection.cs ./TestRadioButton.cs ./TestRange.cs ./TestStatusbar.cs ./TestDialog.cs ./TestFlipping.cs ./TestSizeGroup.cs ./TestCombo.cs ./TestComboBox.cs ./WidgetViewer.cs
./TestCombo.cs(17,14): warning CS0612: `Gtk.Combo' is obsolete
./TestRange.cs(35,6): error CS0104: `Range' is an ambiguous reference between `System.Range' and `Gtk.Range'
@/external/bockbuild/stage/lib/mono/4.5/mscorlib.dll (Location of the symbol related to previous error)
@/external/bockbuild/builds/gtk-sharp-None/sample/test/../../gtk/gtk-sharp.dll (Location of the symbol related to previous error)
./TestRange.cs(35,13): error CS0030: Cannot convert type `Gtk.HScale' to `System.Range'
./TestRange.cs(42,6): error CS0104: `Range' is an ambiguous reference between `System.Range' and `Gtk.Range'
@/external/bockbuild/stage/lib/mono/4.5/mscorlib.dll (Location of the symbol related to previous error)
@/external/bockbuild/builds/gtk-sharp-None/sample/test/../../gtk/gtk-sharp.dll (Location of the symbol related to previous error)
./TestRange.cs(42,13): error CS0030: Cannot convert type `Gtk.HScrollbar' to `System.Range'
./TestRange.cs(62,6): error CS0104: `Range' is an ambiguous reference between `System.Range' and `Gtk.Range'
@/external/bockbuild/stage/lib/mono/4.5/mscorlib.dll (Location of the symbol related to previous error)
@/external/bockbuild/builds/gtk-sharp-None/sample/test/../../gtk/gtk-sharp.dll (Location of the symbol related to previous error)
./TestRange.cs(62,13): error CS0030: Cannot convert type `Gtk.VScale' to `System.Range'
```

This is because the compiler is resolving `Range` in the test as `System.Range`
instead of `Gtk.Range`, so we disambiguate it.

/cc @marek-safar
---
 sample/test/TestRange.cs | 2 ++
 1 file changed, 2 insertions(+)

NB: this is not really a430f0774dcc075110f3f277c9e7e45fd7f6f883; it
has been trivially modified to appease gentoo's epatch

diff -urpN gtk-sharp-2.12.21.orig/sample/test/TestRange.cs gtk-sharp-2.12.21/sample/test/TestRange.cs
--- gtk-sharp-2.12.21.orig/sample/test/TestRange.cs	2012-09-25 14:19:02.000000000 -0700
+++ gtk-sharp-2.12.21/sample/test/TestRange.cs	2019-10-15 11:47:23.850474560 -0700
@@ -9,6 +9,8 @@
 using System;
 
 using Gtk;
+// disambiguate, Gtk.Range vs System.Range
+using Range=Gtk.Range;
 
 namespace WidgetViewer {
 
